cdep/src/main/java/io/cdep/cdep/generator/CMakeGenerator.java

```diff
@@ -319,15 +359,80 @@ private void visit(@NotNull Expression expression) {
               specific.size.toString(),
               specific.sha256));
       if (specific.includePath != null) {
-        append("%starget_include_directories(${target} PRIVATE ", prefix);
-        visit(specific.includePath);
-        append(")\n");
+        if (commonIncludes != null) {
+          commonIncludes.add(specific.includePath);
+        }
+        insertingIndex = sb.length();
+        insertingPrefix = prefix;
       }
 
       for(Expression libraryPath : specific.libraryPaths) {
-        append("%starget_link_libraries(${target} ", prefix);
+        dependsInsert = 0;
+        String fullPath= libraryPath.toString();
+        String libName;
+        if(fullPath.contains(".") && fullPath.contains("/") && fullPath.lastIndexOf(".") - fullPath.lastIndexOf("/") > 0)
+          libName = fullPath.substring(fullPath.lastIndexOf("/")+1,fullPath.lastIndexOf("."));
+        else if(fullPath.contains("//"))
+          libName = fullPath.substring(fullPath.lastIndexOf("/")+1);
+        else
+          libName = fullPath;
+        append("%sif(NOT (TARGET %s))\n", prefix, libName);
+        append("%s  add_library(%s STATIC IMPORTED )\n", prefix, libName);
+        append("%sendif()\n", prefix);
+        append("%sset_target_properties(%s PROPERTIES IMPORTED_LOCATION ", prefix, libName);
         visit(libraryPath);
+        if(specific.includePath != null || !commonIncludes.isEmpty())
+        {
+          if(specific.includePath != null) {
+            append(" INTERFACE_INCLUDE_DIRECTORIES ");
+            visit(specific.includePath);
+          }
+          if(!commonIncludes.isEmpty()) {
+            for(Expression includer : commonIncludes) {
+              append(" INTERFACE_INCLUDE_DIRECTORIES ");
+              visit(includer);
+            }
+          }
+        }
+        if(compilerFeatures != null)
```

Comment:
Please match the convention of putting { on the same line as if statement

Reply:
KNR style or just for if statements?

Reply:
Yeah, all logic blocks not just if (KNR then?)

Reply:
K&R sorry; typo.   I'm assuming you mean just the style of collapsing logic blocks to same line so this:
if ( ... ) {
} else {
}
as opposed to this:
if ( ... )
{
}
else
{
}

Not meaning the full K&R style guide.